app/src/lib/stores/app-store.ts

```diff
@@ -2523,6 +2581,19 @@ export class AppStore extends TypedBaseStore<IAppState> {
       )
     )
 
+    if (shouldPopStash) {
+      const stash = await getLastDesktopStashEntryForBranch(
+        repository,
+        foundBranch.name
+      )
+
+      if (stash !== null) {
+        await this._popStashEntry(repository, stash)
+      } else {
+        log.warn('no stash found that matches')
```

Comment:
:art:  is this important enough to make the user aware of it? Or is a better message here enough for the developers to look over when troubleshooting:

```suggestion
        log.warn(`[AppStore._checkoutBranch] no stash found that matches ${foundBranch.name}`)
```